### PR TITLE
Fix wrong result in create_expand_pow_optimization(). Partially fixes #15335

### LINKED ISSUE #15335: create_expand_pow_optimization() is not robust
Using SymPy 1.3:

```
import sympy as sp
from sympy.codegen.rewriting import create_expand_pow_optimization
expand_opt = create_expand_pow_optimization(7)
a = sp.symbols("a")
expr = a**(5) - a**2
print("should be a**5 - a**2:",expr)
print("should be a*a*a*a*a - a*a:",expand_opt(expr))
expr = a**(-5)
print("should be a**(-5):",expr)
print("should be equal to a**(-5):", expand_opt(expr))
```

Output is:
```
should be a**5 - a**2: a**5 - a**2
should be a*a*a*a*a - a*a: -a**2 + a*a*a*a*a
should be a**(-5): a**(-5)
should be equal to a**(-5): 1
```

Notice that it failed to optimize `-a**2` and more disturbingly `a**(-5)` was evaluated to `1`.

### PR DESCRIPTION
**Pull request by zachetienne**
#### References to other Issues or PRs

Fixes #15335 (partially)

#### Release Notes

* codegen
  * Fix wrong result in create_expand_pow_optimization()

### CODE REVIEW

**Comment by zachetienne** [comment]
"There was an issue with the release notes. Please do not close this pull request; instead edit the description after reading the guide on how to write release notes."

Where is this "description"?

**Comment by bjodah** [comment]
There's a bot which generates release notes based on pull requests, it requires a specific format (it is documented in the link), but I just edited your PR so that it should match that specification so you don't need to worry about it.

Would you be able to add a regression test for this fix here?:
https://github.com/sympy/sympy/blob/master/sympy/codegen/tests/test_rewriting.py#L161

**Comment by bjodah** [comment]
Let me know if you need help with git/github or if you'd prefer that I simply add the test.

**Comment by asmeurer** [comment]
If you have any suggestions on how to improve the wording of the bot let me know.

**Comment by zachetienne** [comment]
@asmeurer I was confused that the bot actually wanted me to add a comment here. The word "description" didn't appear on this page outside the bot's suggestion.

> No release notes were found. Please add release notes to the pull request description.

Maybe 

"No release notes were found. Please add them as a comment below using the standard format provided on [this page](link to page)"

would be less confusing.

**Comment by zachetienne** [comment]
@bjodah : I have added a test case to check against regressions for this issue: https://github.com/zachetienne/sympy/commit/1e938d64a1947006cbdcc0ab6facb0b2439edcf1 Can this be incorporated into this pull request, or does it need another pull request? I'm not sure how to incorporate into this request.

File: sympy/codegen/tests/test_rewriting.py [hunk 1]
```diff
@@ -169,3 +169,5 @@ def test_create_expand_pow_optimization():
 
     sin4x = sin(x)**4
     assert ccode(optimize(sin4x, [my_opt])) == 'pow(sin(x), 4)'
+
+    assert ccode(optimize((x**(-4)), [my_opt])) == 'x**(-4)'
```
Comment (bjodah) [inline sympy/codegen/tests/test_rewriting.py]: `x**(-4)` is not valid C code...

File: sympy/codegen/tests/test_rewriting.py [hunk 2]
```diff
@@ -169,3 +169,5 @@ def test_create_expand_pow_optimization():
 
     sin4x = sin(x)**4
     assert ccode(optimize(sin4x, [my_opt])) == 'pow(sin(x), 4)'
+
+    assert ccode(optimize((x**(-4)), [my_opt])) == 'pow(x, -4)'
```
Comment (bjodah) [inline sympy/codegen/tests/test_rewriting.py]: OK, that looks better. Eventually we should probably see if we can make it output `1.0/(x*x*x*x)` for this case. But at least this form is not incorrect.

**Comment by bjodah** [comment]
Sorry, missed your question but I see that you've added the commit. Thank you.
I am +1 to merge if tests pass. This would fix the incorrect results. Ideally this can then be improved in subsequent pull requests.

**Comment by bjodah** [comment]
Thank you @zachetienne for this fix.

**Pull request merged**